temoa/components/time.py

```
@@ -82,41 +85,46 @@ def validate_time(M: 'TemoaModel'):
 
         if expected_keys != keys:
-            extra = keys.difference(expected_keys)
-            missing = expected_keys.difference(keys)
-            msg = (
+            extra: set[tuple[int, str, str]] = keys.difference(expected_keys)
+            missing: set[tuple[int, str, str]] = expected_keys.difference(keys)
+            msg: str = (
                 'TimeSegmentFraction elements for period {} do not match TimeSeason and TimeOfDay.'
                 '\n\nIndices missing from TimeSegmentFraction:\n{}'
                 '\n\nIndices in TimeSegmentFraction missing from TimeSeason/TimeOfDay:\n{}'
             ).format(p, missing, extra)
```

Ruff (0.14.0)

Use f-string instead of `format` call

Convert to f-string

(UP032)